src/Block/NonceFieldBlock.php

```diff
@@ -36,7 +41,18 @@ public function __construct(
     ) {
         $this->_setNonce($nonce)
             ->_setFieldName($fieldName)
-            ->_setUseReferer($useReferer);
+            ->_setUseReferer($useReferer)
+            ->_setRefererFieldName(static::REFERER_FIELD_NAME);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _getRefererUrl()
+    {
+        return \esc_attr(\wp_unslash($_SERVER['REQUEST_URI']));
```

Comment:
@XedinUnknown What do you think about removing the `$useReferer` flag from `AbstractNonceFieldBlock` and adding a `$referer` property, which if `null` results in the block not using the referer? It makes things more concise and the constructor for the `NonceFieldBlock` cleaner by simply having the `$referer` default to `null`.